Source/Extensions/Result.swift

```diff
@@ -4,9 +4,10 @@ import Result
 public extension Result {
   static func fromDecoded<T>(decoded: Decoded<T>) -> Result<T, NSError> {
     switch decoded {
-    case let .Success(obj): return .Success(obj)
-    case .TypeMismatch, .MissingKey:
-      return .Failure(error(decoded.description))
+    case let .Success(obj):
+      return .Success(obj)
+    case let .Failure(decodedError):
+      return .Failure(error(decodedError.description))
```

Comment:
Doing this will cause us to have to change `NSError` to `DecodeError` all the way down (e.g. in the `Request` protocol itself.) We're going to merge with the cast to `NSError` and revisit at a later date.